bootstrap/kubeadm/internal/cloudinit/cloudinit_test.go

```diff
@@ -34,6 +34,7 @@ func TestNewInitControlPlaneAdditionalFileEncodings(t *testing.T) {
 	cpinput := &ControlPlaneInput{
 		BaseUserData: BaseUserData{
 			Header:              "test",
+			BootCommands:        nil,
```

Comment:
Can we have a test for BootCommands not nil for init, join CP, join workers?

Reply:
I added non-nil values for `TestNewInitControlPlaneCommands` and `TestNewJoinControlPlaneAdditionalFileEncodings`, did you want me to add more tests or is this enough? I couldn't find one for "join workers" https://github.com/kubernetes-sigs/cluster-api/pull/11271/commits/daf4f97e56614c62bec80c80635f094a81cfca33